shell/platform/linux/public/flutter_linux/fl_platform_views.h

```diff
@@ -0,0 +1,224 @@
+ *                                         FlValue *args) {
+ *     // Creates a new instance of your custom #FlPlatformView implementation.
+ *     // Flutter will take the reference, and release it when Dart-side widget
+ *     // is destructed.
+ *     WebViewFactory *self = WEBVIEW_FACTORY (factory);
+ *     return FL_PLATFORM_VIEW (
+ *         webview_platform_view_new (self->messenger, view_identifier, args));
+ *   }
+ *
+ *   static FlMessageCodec *
+ *   webview_factory_get_create_arguments_codec (FlPlatformViewFactory *self) {
+ *     // Simply creates a new instance of message codec.
+ *     return FL_MESSAGE_CODEC (fl_standard_message_codec_new ());
+ *   }
+ *
+ *   static void
+ *   webview_factory_fl_platform_view_factory_iface_init (
+ *     FlPlatformViewFactoryInterface *iface) {
+ *     iface->create_platform_view = webview_factory_create_platform_view;
+ *     iface->get_create_arguments_codec =
+ *         webview_factory_get_create_arguments_codec;
+ *   }
+ * ]|
+ */
+
+struct _FlPlatformViewFactoryInterface {
+  GTypeInterface parent_interface;
+
+  /**
+   * FlPlatformViewFactory::create_platform_view:
+   * @factory: an #FlPlatformViewFactory.
+   * @view_identifier: a unique identifier for this #GtkWidget.
+   * @args: parameters for creating the #GtkWidget sent from the Dart side of
+   *        the Flutter app. If get_create_arguments_codec is not implemented,
+   *        or if no creation arguments were sent from the Dart code, this will
+   *        be null. Otherwise this will be the value sent from the Dart code as
+   *        decoded by get_create_arguments_codec.
+   *
+   * Creates an #FlPlatformView for embedding.
+   *
+   * Returns: (transfer full): an #FlPlatformView.
+   */
+  FlPlatformView* (*create_platform_view)(FlPlatformViewFactory* factory,
+                                          int64_t view_identifier,
+                                          FlValue* args);
+
+  /**
+   * FlPlatformViewFactory::get_create_arguments_codec:
+   * @factory: an #FlPlatformViewFactory.
+   *
+   * Creates an #FlMessageCodec if needed. Only needs to be implemented if
+   * create_platform_view needs arguments.
+   *
+   * Returns: (transfer full): an #FlMessageCodec for decoding the args
+   * parameter of create_platform_view, or %NULL if no creation arguments will
+   * be sent.
+   */
+  FlMessageCodec* (*get_create_arguments_codec)(FlPlatformViewFactory* self);
+};
+
+/**
+ * fl_platform_view_factory_create_platform_view:
+ * @factory: an #FlPlatformViewFactory.
+ * @view_identifier: a unique identifier for this #GtkWidget.
+ * @args: parameters for creating the #GtkWidget sent from the Dart side of
+ *        the Flutter app. If get_create_arguments_codec is not implemented, or
+ *        if no creation arguments were sent from the Dart code, this will be
+ *        null. Otherwise this will be the value sent from the Dart code as
+ *        decoded by get_create_arguments_codec.
```

Comment:
These last two lines don't make sense here (copy/paste from above). They can be removed.

Reply:
I think that we can hide `fl_platform_view_factory_create_platform_view` from plugins, and can be moved to `fl_platform_views_private.h`.